### PR TITLE
Don't remove MIR storage markers in debug

### PR DESCRIPTION
**Pull request by Noratrieb**
These markers are pretty good at detecting compiler bugs during MIR validation. We don't want to emit LLVM lifetime storage annotations in unoptimized builds so we just filter them out at the codegen stage.

### CODE REVIEW

**Comment by Nilstrieb** [comment]
looks like we're trying really hard now

File: compiler/rustc_codegen_ssa/src/mir/mod.rs [hunk 1]
```diff
@@ -99,6 +99,10 @@ pub struct FunctionCx<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> {
 
     /// Caller location propagated if this function has `#[track_caller]`.
     caller_location: Option<OperandRef<'tcx, Bx::Value>>,
+
+    /// Whether `StorageLive` and `StorageDead`.should be ignored.
```
Comment (bjorn3) [inline compiler/rustc_codegen_ssa/src/mir/mod.rs:103]:
```suggestion
    /// Whether `StorageLive` and `StorageDead` should be ignored.
```

**Comment by Nilstrieb** [comment]
This would be neat, but I don't think the perf regressions are worth it.

**Pull request closed without merging**